src/connection_pool.rs

```diff
@@ -14,69 +14,90 @@ use tokio::sync::RwLock;
 // Pool for keeping open connections. Pooled connections are associated with a `ConnectionRemover`
 // which can be used to remove them from the pool.
 #[derive(Clone)]
-pub(crate) struct ConnectionPool {
-    store: Arc<RwLock<Store>>,
+pub(crate) struct ConnectionPool<T: Id> {
```

Comment:
Nitpick: we probably should be consistent with the type parameter name. We're mostly using `I: Id`, which I think is a bit better naming.